examples/dcn.py

```diff
@@ -0,0 +1,522 @@
+    "num_epochs": 100,
+    "batch_size": 1024,
+}
+
+MOVIELENS_CONFIG = {
+    # features
+    "int_features": [
+        "movie_id",
+        "user_id",
+        "user_gender",
+        "bucketized_user_age",
+    ],
+    "str_features": [
+        "user_zip_code",
+        "user_occupation_text",
+    ],
+    # model
+    "embedding_dim": 32,
+    "deep_net_num_units": [192, 192, 192],
+    "projection_dim": 20,
+    "dcn_num_units": [192, 192],
+    # training
+    "learning_rate": 0.4,
+    "num_epochs": 10,
+    "batch_size": 1024,
+}
+
+LOOKUP_LAYERS = {
+    "int": keras.layers.IntegerLookup,
+    "str": keras.layers.StringLookup,
+}
+
+"""
+Here, we define a helper function for visualising weights of the cross layer in
+order to better understand its functioning. Also, we define a function for
+compiling, training and evaluating a given model.
+"""
+
+
+def visualize_layer(matrix, features):
+    plt.figure(figsize=(9, 9))
+
+    im = plt.matshow(np.abs(matrix), cmap=plt.cm.Blues)
+
+    ax = plt.gca()
+    divider = make_axes_locatable(plt.gca())
+    cax = divider.append_axes("right", size="5%", pad=0.05)
+    plt.colorbar(im, cax=cax)
+    cax.tick_params(labelsize=10)
+    ax.set_xticklabels([""] + features, rotation=45, fontsize=10)
+    ax.set_yticklabels([""] + features, fontsize=10)
+
+
+def train(learning_rate, epochs, train_data, test_data, model, model_name=None):
+    optimizer = keras.optimizers.AdamW(learning_rate=learning_rate)
+    loss = keras.losses.MeanSquaredError()
+    rmse = keras.metrics.RootMeanSquaredError()
+
+    model.compile(
+        optimizer=optimizer,
+        loss=loss,
+        metrics=[rmse],
+    )
+
+    if model_name is not None:
+        print("\n=== Training", model_name, "===\n")
+    model.fit(
+        train_data,
+        epochs=epochs,
+        verbose=0,
+    )
+
+    if model_name is not None:
+        print("\n=== Evaluating", model_name, "===\n")
+    results = model.evaluate(test_data, return_dict=True, verbose=0)
+    rmse_value = results["root_mean_squared_error"]
+    num_params = model.count_params()
+    print(f"RMSE: {rmse_value:.4f}, #params: {num_params}")
+
+
+"""
+## Toy Example
+
+To illustrate the benefits of DCNs, let's consider a simple example. Suppose we
+have a dataset for modeling the likelihood of a customer clicking on a blender
+advertisement. The features and label are defined as follows:
+
+| **Features / Label** | **Description**                | **Range** |
+|:--------------------:|:------------------------------:|:--------:|
+| $x_1$ = country      | Customer's resident country    | [0, 199] |
+| $x_2$ = bananas      | # bananas purchased            | [0, 23]  |
+| $x_3$ = cookbooks    | # cooking books purchased      | [0, 5]   |
+| y                    | Blender ad click likelihood    | -        |
+
+Then, we let the data follow the following underlying distribution:
+$y = f(x_1, x_2, x_3) = 0.1x_1 + 0.4x_2 + 0.7x_3 + 0.1x_1x_2 +$  
+$3.1x_2x_3 + 0.1x_3^2$.
```

Comment:
I don't think this $ thing works on keras.io. Can you double check?

Reply:
I saw it being used in other examples: https://github.com/keras-team/keras-io/blob/master/examples/nlp/multiple_choice_task_with_transfer_learning.py#L158

Reply:
Oh, but it hasn't been rendered properly here: https://keras.io/examples/nlp/multiple_choice_task_with_transfer_learning/. Cool, will get rid of it.

Reply:
https://github.com/keras-team/keras-io/blob/master/contributor_guide.md#mathematical-notation

Reply:
Yeah, it's unfortunate. I'm not sure what this standard is, it's not markdown. Maybe there is a plugin to add support for it.

Reply:
The solution right now is to add images.

Reply:
Added it inside backticks. It won't be rendered as a LaTeX equation, but something's better than nothing